codegen/src/cargo_toml.rs

```diff
@@ -24,13 +24,15 @@ version = "0.1.0"
 edition = "2018"
 
 [dependencies]
+azure_core = {{ path = "../../../sdk/core", version = "0.1.0" }}
```

Comment:
We shouldn't be relying on people having the sdk repo on their machine. We can use git instead for now.

Reply:
I do not understand. These examples are in the git repo and need to build against the other projects. I'm a little confused why `version` is used.

Reply:
We can put info in the readme about how to use these dependencies currently as git dependencies.

Reply:
This requires that builders of this repo have the sdk repo cloned on their machine which I don't believe we should require. If we change azure_core to be a git dependency we can avoid this for now. Ultimately we'll want to change this to a normal crates.io dependency once that crate is published.

Reply:
The line you are looking at here is to produce a `Cargo.toml` file. It is not a dependency of this repository. It is easier to see if you look at it like this:

<img width="805" alt="image" src="https://user-images.githubusercontent.com/80104/97178674-6e02b280-175d-11eb-80ea-9843c53e6aea.png">

Regarding what it produces, it is doing the right thing by specifying a path and a version. The path is works locally when you clone the sdk. It works when you use a git dependency. I just validated this this works:

``` toml
[dependencies]
azure_mgmt_storage = { branch = "TokenCredential", git = "https://github.com/ctaggart/azure-sdk-for-rust" }
azure_identity = { branch = "TokenCredential", git = "https://github.com/ctaggart/azure-sdk-for-rust" }
tokio = { version = "*", features = ["macros"] }
reqwest = { version = "*", features = ["json"] }
```

It will use the version dependency when published to a registry like crates.io according to https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#multiple-locations.

Reply:
Yes, to produce a `Cargo.toml` for non `azure-sdk-for-rust` repos, it would need to be a git dependency. That can be coded later. For now, `azure-sdk-for-rust` is my only target for this.

Reply:
Ah, so sorry. Carry on!